File: maximumSubarray/maximumSubarray.js
// https://leetcode.com/problems/maximum-subarray
/**
 * @param {number[]} nums
 * @return {number}
 */
const maxSubArray = (nums) => {
    let bestSum = nums[0];
    let currentSum = nums[0];
    const length = nums.length;
    for (let i = 1; i < length; i++) {
        currentSum = Math.max(nums[i], currentSum + nums[i]);
        bestSum = Math.max(bestSum, currentSum);
    }
    return bestSum;
};

File: maximumSubarray/maximumSubarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        best_sum = nums[0]  # or: float('-inf')
        current_sum = nums[0]
        for i in range(1, len(nums)):
            current_sum = max(nums[i], current_sum + nums[i])
            best_sum = max(best_sum, current_sum)
        return best_sum
